core/src/main/java/com/sequenceiq/cloudbreak/service/sharedservice/DatalakeService.java

```diff
@@ -0,0 +1,268 @@
+        Optional<Stack> datalakeStack = Optional.ofNullable(datalakeResources)
+                .map(Set::stream).flatMap(Stream::findFirst)
+                .map(DatalakeResources::getDatalakeStackId)
+                .map(stackService::getByIdWithListsInTransaction);
+
+        return datalakeStack;
+    }
+
+    public Long getDatalakeResourceId(Long datalakeStackId) {
+        Optional<DatalakeResources> datalakeResources = datalakeResourcesService.findByDatalakeStackId(datalakeStackId);
+        if (datalakeResources.isPresent()) {
+            return datalakeResources.get().getId();
+        }
+        return null;
+    }
+
+    public SharedServiceConfigsView createSharedServiceConfigs(Stack source) {
+        Cluster cluster = source.getCluster();
+        Optional<DatalakeResources> datalakeResources = getDataLakeResource(source);
+        return createSharedServiceConfigs(cluster.getBlueprint(), cluster.getPassword(), datalakeResources);
+    }
+
+    public String getDatalakeCrn(StackV4Request source, Workspace workspace) {
+        if (source.getSharedService() != null && isNotBlank(source.getSharedService().getDatalakeName())) {
+            Optional<Stack> result = stackService
+                    .findStackByNameOrCrnAndWorkspaceId(NameOrCrn.ofName(source.getSharedService().getDatalakeName()), workspace.getId());
+            if (result.isPresent()) {
+                return result.get().getResourceCrn();
+            } else {
+                LOGGER.debug("No datalake resource found for data lake: {}", source.getSharedService().getDatalakeName());
+            }
+        }
+        return null;
+    }
+
+    public void setDatalakeIdOnStack(Stack stack, StackV4Request source, Workspace workspace) {
+        DatalakeResources datalakeResources = getDatalakeResourceId(source, workspace);
+        if (datalakeResources != null) {
+            stack.setDatalakeResourceId(datalakeResources.getId());
+            stack.setDatalakeCrn(stackService.get(datalakeResources.getDatalakeStackId()).getResourceCrn());
+        } else {
+            stack.setDatalakeResourceId(null);
+        }
+    }
+
+    private Optional<DatalakeResources> getDataLakeResource(Stack source) {
+        if (source.getDatalakeResourceId() != null) {
+            return datalakeResourcesService.findById(source.getDatalakeResourceId());
+        }
+        return Optional.empty();
+    }
+
+    private String getHostForComponentInService(DatalakeResources datalakeResources, String service, String serviceComponent) {
+        if (datalakeResources.getServiceDescriptorMap().containsKey(service)) {
+            ServiceDescriptor serviceDescriptor = datalakeResources.getServiceDescriptorMap().get(service);
+            Map<String, Object> componentHostMap = serviceDescriptor.getComponentsHosts().getMap();
+            return String.valueOf(componentHostMap.get(serviceComponent));
+        } else {
+            return null;
```

Comment:
Like in the case of other methods it would be nice to return an optional instead of null just to warn the developer about that this value can be null and should be handled that way later on